health/service/src/main/java/com/cerner/beadledom/health/HttpHealthDependency.java

```diff
@@ -0,0 +1,54 @@
+package com.cerner.beadledom.health;
+
+import java.io.IOException;
+import java.util.Objects;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Implementation of {@link HealthDependency} to check the health of any HTTP service.
+ *
+ * @author Venkatesh Sridharan.
+ */
+public class HttpHealthDependency extends HealthDependency {
+  private final String availabilityUrl;
+  private final String name;
+  private final OkHttpClient client;
+
+  /**
+   * Constructs an instance of {@link HttpHealthDependency} with the given parameters.
+   *
+   * @param availabilityUrl String representing URL of the availability health check.
+   * @param name String representing name of the dependency
+   * @param isPrimary Boolean representing whether the dependency is primary or not.
+   */
+  public HttpHealthDependency(String availabilityUrl, String name, boolean isPrimary) {
+    super(isPrimary);
+    Objects.requireNonNull(availabilityUrl);
+    Objects.requireNonNull(name);
+
+    this.availabilityUrl = availabilityUrl;
+    this.name = name;
+    this.client = new OkHttpClient.Builder().build();
+  }
+
+  @Override
+  public HealthStatus checkAvailability() {
+    Request request = new Request.Builder().url(availabilityUrl).head().build();
```

Comment:
Possibly stupid question..do we not configure retries for health checks?

Reply:
hmm, I have not seen it being done in the past. Do others have thoughts on this?

Reply:
OkHttp already defaults to retry on connection errors. I think that is all that would be required for health checks.

Reply:
I don't think the Ruby one does retries so I think that's the way to go here for now. Especially considering the architectures group discussions around potentially completely revamping how we do health checks.